tests/unit/test_common.cpp

```diff
@@ -415,3 +416,27 @@ TEST_F(SerializerTest, HNSWSerialzer) {
     ASSERT_EXCEPTION_MESSAGE(HNSWFactory::NewIndex(this->file_name), std::runtime_error,
                              "Cannot load index: bad algorithm type");
 }
+
+void test_log_impl(void *ctx, const char *message) {
+    std::vector<std::string> *log = (std::vector<std::string> *)ctx;
+    log->push_back(message);
+}
```

Comment:
We plan on setting the index name in the `logCtx`, and having the callback prepend it to the message right? Consider simulating this scenario in this test as well...